guide/additional-info/async-await.md

```diff
@@ -0,0 +1,163 @@
+# Understanding async/await
+
+If you aren't very familiar with ECMAScript 2017, you may not know about async/await. It's a useful way to handle Promises in a synchronous manner instead stacking `.then()` callbacks. Also, it looks cleaner and is increases readability overall.
+
+## How do Promises work?
+
+Before we can get into async/await, you should know what Promises are and how they work, because async/await is just a way to handle Promises. if you know "what Promises are and how to deal with them" you can skip this part. 
+
+Promises are a way to handle asynchronous tasks in Javascript; they are the newer alternative to callbacks. A Promise has a lot of similarities to a progress bar; Promises represent an ongoing process that has not yet finished. A good example for that is a request to a server (e.g discord.js sends requests to Discord's API).
+
+A Promise can have 3 states pending, fulfilled, and rejected
+
+The state **pending** means that the Promise still is ongoing and nether fulfilled nor rejected.
+The state **resolved** means that the Promise is done and was executed without any errors.
+The state **rejected** means that the Promise encountered an error and could not be executed correctly.
+
+One important thing to know is that a Promise can only have one state at a time; it can never be pending and fulfilled, rejected and fulfilled, or pending and rejected. You may be asking, "How would that look in code?". Here is a small example:
+<p class="tip">ES6 code is being used in this example; if you do not know what that is, you should read up on that [here](/additional-info/es6-syntax).</p>
+
+```js
+function asyncTask() {
+	return Promise((resolve, reject) => {
```

Comment:
Missing the `new` before `Promise`; this is currently returning `TypeError: undefined is not a promise`. However, you're also not using `reject` anywhere in here, which I feel like makes this a bit redundant.

Mostly just a suggestion, but why not try to create a real world example instead of pseudo code? I've noticed that real world examples help people learn more instead of just giving them the syntax and hoping for the best.